### ISSUE TITLE
RFT problem

### ISSUE THREAD

**Issue by Dorothylyly** [OpenRLHF/OpenRLHF#786]
when I follow the RFT demo to build my own pipline，

<img width="871" alt="Image" src="https://github.com/user-attachments/assets/de23e518-a45e-4fe3-a1ea-1a2e36855f23" />

I find this ：

<img width="824" alt="Image" src="https://github.com/user-attachments/assets/5d735ffb-6fb0-480d-a278-e12befe2314c" />

Do I have to build a “http”-serve？

my start.sh like：

deepspeed --module openrlhf.cli.train_ppo \
  --pretrain pretrain_models/LLM-Research/Meta-Llama-3-8B-Instruct \
  --**remote_rm_url OpenRLHF/RFT/reward_func.py** \
  --save_path ./checkpoint/llama-3-8b-rlhf \
  --save_steps -1 \
  --logging_steps 1 \
  --eval_steps -1 \
  --micro_train_batch_size 2 \
  --train_batch_size 128 \
  --micro_rollout_batch_size 4 \
  --rollout_batch_size 1024 \
  --max_epochs 1 \
  --prompt_max_len 1024 \
  --generate_max_len 1024 \
  --zero_stage 2 \
  --bf16 \
  --actor_learning_rate 5e-7 \
  --critic_learning_rate 9e-6 \
  --init_kl_coef 0.01 \
  --prompt_data ./OpenRLHF \
  --input_key context_messages \
  --apply_chat_template \
  --max_samples 100000 \
  --normalize_reward \
  --adam_offload \
  --flash_attn \
  --gradient_checkpointing \

**Comment by 2proveit** [comment]
this should be your local Python file.

**Comment by Dorothylyly** [comment]
@2proveit its true，i do give my local python file path，but still get such error，here is my local path：--remote_rm_url OpenRLHF/RFT/reward_func.py。seems it will call the remote API whatever

**Comment by 2proveit** [comment]
@Dorothylyly I think you may double-check the path to the file, since you are running this command in the path `OpenRLHF`, you may remove the OpenRLHF prefix, or you can add 
```
if __name__ == "__main__":
```
 to you `reward_func.py`.
 The simplest way is to use an absolute path to ensure you have passed the correct file path.

**Comment by Dorothylyly** [comment]
@2proveit heres my absolute path，“/workspace/OpenRLHF/RFT/reward_func.py”， and it still reports：
please check: Invalid URL '/workspace/OpenRLHF/RFT/reward_func.py': No scheme supplied. Perhaps you meant https:///workspace/OpenRLHF/RFT/reward_func.py'?

my reward_func.py:

<img width="504" alt="Image" src="https://github.com/user-attachments/assets/707795c9-349c-4774-b316-6ba485f0106e" />

**Comment by 2proveit** [comment]
@Dorothylyly the reward function loading code is located at [https://github.com/OpenRLHF/OpenRLHF/blob/273422305ea17362319f5569c6f9ef5a16b49cb0/openrlhf/trainer/ppo_utils/experience_maker.py#L159](https://github.com/OpenRLHF/OpenRLHF/blob/273422305ea17362319f5569c6f9ef5a16b49cb0/openrlhf/trainer/ppo_utils/experience_maker.py#L159)
it may help you

**Comment by Dorothylyly** [comment]
@2proveit hi maybe i found a bug:
heres the entrance of the NaiveExperienceMaker: https://github.com/OpenRLHF/OpenRLHF/blob/main/openrlhf/trainer/ppo_trainer.py#L141
<img width="383" alt="Image" src="https://github.com/user-attachments/assets/9dfce25f-262c-40b2-bf18-78583e246ce1" />
the type of "remote_rm_url" here is str, but in the __init__ function of NaiveExperienceMaker, the type is changed:

<img width="532" alt="Image" src="https://github.com/user-attachments/assets/0b46aaf6-0c88-430c-90b9-783c976414d2" />

<img width="841" alt="Image" src="https://github.com/user-attachments/assets/a4877cc2-85be-468f-84ad-3ec94e31f3fb" />
then the string[0] is absolutely a "char"?
